src/multio/action/scale/Mapping.cc

```diff
@@ -0,0 +1,51 @@
+#include "Mapping.h"
+
+#include <cstdlib>
+#include <string>
+
+#include "multio/LibMultio.h"
+#include "multio/message/Glossary.h"
+#include "multio/util/Substitution.h"
+
+
+namespace multio::action {
+
+using message::glossary;
+
+ScaleMapping::ScaleMapping(const config::ComponentConfiguration& compConf) : hasMapping_(false), scaleMap_{} {
+
+    const auto mappings = compConf.parsedConfig().has("mapping-definition")
+                            ? compConf.parsedConfig().getSubConfigurations("mapping-definition")
+                            : std::vector<eckit::LocalConfiguration>{};
+
+    if (!mappings.empty()) {
+        hasMapping_ = true;
+        for (const auto& mapping : mappings) {
+            auto matcher = mapping.getSubConfiguration("case");
+            scaleMap_[matcher.getString("param-is")] = matcher.getString("map-to-param");
+        }
+    }
+}
+
+void ScaleMapping::applyMapping(message::Metadata& md) const {
+    if (hasMapping_) {
+        std::string cparam{"xxx"};
```

Comment:
Same comments here about the logic and duplication (triplication). I'm a bit confused. I think we need this in two places but not in three?

Reply:
see explanation above. Happy to change but would like to discuss what makes the most sense